experiment_prototype/experiment_prototype.py

```diff
@@ -732,7 +759,107 @@ def slice_beam_directions_mapping(self, slice_id):
             mapping[beam_num] = beam_dir
         return mapping
 
-    def add_slice(self, exp_slice, interfacing_dict=None):
+    def check_new_slice_interfacing(interfacing_dict):
```

Comment:
so in terms of why it needed to change, the old method was quite messy (didn't require interfacing info on addition). This is a problem for writing slice interfacing to file because if we are writing to file we don't want the user to change after the addition, it has to be only set once. `check_new_slice_interfacing` replaces `check_interfacing` (from the bottom of the file). It also does a much better check that should prevent failure of interfacing within the scan classes. We weren't checking that the interfacing made sense before, only that it existed. 
Also, the check is now done in add_slice when added; previously check_interfacing was only called later in build_scans, when the scan classes are being created. 
I can make a diagram for the docs sometime.